### ISSUE TITLE
Adjust frontends to new ARGUMENT edge

### ISSUE THREAD

**Issue by ml86** [ShiftLeftSecurity/codepropertygraph#462]
In this PR https://github.com/ShiftLeftSecurity/codepropertygraph/pull/461 i introduced a the new ARGUMENT edge in the usual backwards compatible manner with a compatibility pass which handles old format CPGs.

ARGUMENT edges need to be present between CALL nodes and their arguments and RETURN nodes and their returnExpression. The ARGUMENT do not replace the current AST edges which stay as they are.

Why we need the ARGUMENT edges:
So far the arguments of a CALL node where defined via its AST children. This is not possible anymore for constructs like function pointer calls where the receiver is not an argument to the called function. Thus the arguments needed an explicit representation in the graph. E.g. C call `funcPtr(a)`: `funcPtr` is the receiver but not an argument to the called function. `a` is the argument and both `funcPtr` and `a` are AST children of the CALL node.
For the RETURN node this problem did not occur so we could have stayed with the AST edge but to keep things homogen I also added the addition ARGUMENT edge requirement there so that one always finds the instruction using an argument via the ARGUMENT edge.

Let me know if this is a problem for one of the languages we support or if you see other problems with this or have questions regarding the format change.
The following list shows who adjusts which frontend:

**Comment by ltcmelo** [comment]
This is a mess-up. We're attempting to fix a flawed design by means of another flawed design, instead of addressing what is wrong in the first place (ending up with a complex and nonorthodox code representation).

We don't need the ARGUMENT edges. The conceptual problem here is that we don't represent, for a call expression, its  **immediate base expression**. That should (since the beginning - see observation below) be done. With the structure from the _complete expression_, a traversal could simply skip the first AST child if only the arguments are of interested.

Observation: In the early phases of _csharp2cpg_, I was preserving such immediate base expression of call expressions. To my surprise, this wasn't expected in the data flow logic. I raised the point to you @ml86, and, as usually happened, I had to work around it -- there's even a [comment in the code explaining this awkward case](https://github.com/ShiftLeftSecurity/csharp2cpg/blob/318195ea424a47108583dff5748b7cd8da03f88c/analysis/io/shiftleft/net2cpg/frontend/CsAstVisitor.cs#L122). But, as it also usually happens, when you bump yourself into the problem, then the CPG format must be changed. 

cc: @fabsx00
